src/garage/tf/regressors/bernoulli_mlp_regressor.py

```diff
@@ -1,167 +1,242 @@
-            self.name = name
-
-            self.normalize_inputs = normalize_inputs
-            self.x_mean_var = x_mean_var
-            self.x_std_var = x_std_var
-            self.first_optimized = not no_initial_trust_region
+                network_output=[y_hat],
+                inputs=[input_var, ys_var, old_prob_var],
+                leq_constraint=(mean_kl, self._max_kl_step))
+            self._first_optimized = False
 
     def fit(self, xs, ys):
-        if self.normalize_inputs:
+        """
+        Fit with input data xs and label ys.
+
+        Args:
+            xs (numpy.ndarray): Input data.
+            ys (numpy.ndarray): Label of input data.
+        """
+        if self._normalize_inputs:
             # recompute normalizing constants for inputs
-            new_mean = np.mean(xs, axis=0, keepdims=True)
-            new_std = np.std(xs, axis=0, keepdims=True) + 1e-8
-            tf.compat.v1.get_default_session().run(
-                tf.group(
-                    tf.compat.v1.assign(self.x_mean_var, new_mean),
-                    tf.compat.v1.assign(self.x_std_var, new_std),
-                ))
-            # self._x_mean_var.set_value(np.mean(xs, axis=0, keepdims=True))
-            # self._x_std_var.set_value(
-            #     np.std(xs, axis=0, keepdims=True) + 1e-8)
-        if self.use_trust_region and self.first_optimized:
-            old_p = self.f_p(xs)
-            inputs = [xs, ys, old_p]
-            optimizer = self.tr_optimizer
+            self.model.networks['default'].x_mean.load(
+                np.mean(xs, axis=0, keepdims=True))
+            self.model.networks['default'].x_std.load(
+                np.std(xs, axis=0, keepdims=True) + 1e-8)
+
+        if self._use_trust_region and self._first_optimized:
+            # To use trust region constraint and optimizer
+            old_prob = self._f_prob(xs)
+            inputs = [xs, ys, old_prob]
+            optimizer = self._tr_optimizer
         else:
             inputs = [xs, ys]
-            optimizer = self.optimizer
+            optimizer = self._optimizer
         loss_before = optimizer.loss(inputs)
-        if self.name:
-            prefix = self.name + '/'
-        else:
-            prefix = ''
-        tabular.record(prefix + 'LossBefore', loss_before)
+        tabular.record('{}/LossBefore'.format(self._name), loss_before)
         optimizer.optimize(inputs)
         loss_after = optimizer.loss(inputs)
-        tabular.record(prefix + 'LossAfter', loss_after)
-        tabular.record(prefix + 'dLoss', loss_before - loss_after)
-        self.first_optimized = True
+        tabular.record('{}/LossAfter'.format(self._name), loss_after)
+        tabular.record('{}/dLoss'.format(self._name), loss_before - loss_after)
+        self._first_optimized = True
 
     def predict(self, xs):
-        return self.f_predict(np.asarray(xs))
+        """
+        Predict ys based on input xs.
 
-    def sample_predict(self, xs):
```

Comment:
Please add back `sample_predict` and `predict_log_likelihood` in the model version, and tests for the functions.